packages/cli/src/utils/deploy-v2.ts

```diff
@@ -171,8 +199,34 @@ export async function deploy(mudConfig: MUDConfig, deployConfig: DeployConfig):
     );
   }
 
-  // Await all promises
-  await Promise.all(promises);
+  // Wait for access to be granted before installing modules
+  await Promise.all(promises); // ----------------------------------------------------------------------------------------------
+
+  // Install modules
+  promises.push(
+    ...mudConfig.modules.map(async (module) => {
```

Comment:
haha, this was an interview question for me at Stripe! this can actually fail in JS because the list of arguments can get too long

should use `splice` instead